changelog/2058.feature.rst

```diff
@@ -0,0 +1 @@
+Added a 'notches' keyword option to thomson.spectral_density, which allows users to notch out the output spectrum over a wavelength range. Also made some changes to the thomson fitting models to include the notch as an option.
```

Comment:
thomson -> Thomson

Maybe rephrase last sentence 

"Changed the `~plasmapy.diagnostics.thomson.spetctral_density_model` function to allow a notch to be applied during fitting."


Right now a notch is added manually in the `run_fit()` function in the Thomson test file. Step one of testing  this PR would be to replace that manual implementation with your new implementation and make sure all the tests pass.